Examples/Nodify.Shapes/Canvas/CanvasView.xaml

```diff
@@ -0,0 +1,542 @@
+
+                <DataTemplate DataType="{x:Type local:TriangleViewModel}">
+                    <Polygon Points="0,100 50,0 100,100"
+                             Stretch="Fill"
+                             Fill="{Binding Color, Converter={StaticResource ColorToSolidColorBrushConverter}}"
+                             Stroke="{Binding BorderColor, Converter={StaticResource ColorToSolidColorBrushConverter}}"
+                             StrokeThickness="2"
+                             Opacity="0.8" />
+                </DataTemplate>
+
+                <DataTemplate DataType="{x:Type local:ShapeToolbarViewModel}">
+                    <Canvas Visibility="{Binding Shape, Converter={StaticResource BooleanToVisibilityConverter}}">
+                        <shared:Swatches SelectedColor="{Binding Shape.Color}"
+                                         Colors="{x:Static local:ShapeViewModel.Colors}"
+                                         Canvas.Top="-70"
+                                         Panel.ZIndex="1">
+                            <shared:Swatches.Effect>
+                                <DropShadowEffect ShadowDepth="1" />
+                            </shared:Swatches.Effect>
+                        </shared:Swatches>
+                    </Canvas>
+                </DataTemplate>
+
+                <DataTemplate DataType="{x:Type local:UserCursorViewModel}">
+                    <StackPanel IsHitTestVisible="False">
+                        <Viewbox Width="24"
+                                 Height="24"
+                                 Margin="-10 0 0 5"
+                                 Stretch="Fill"
+                                 HorizontalAlignment="Left">
+                            <Path Fill="{Binding Color, Converter={StaticResource ColorToSolidColorBrushConverter}}"
+                                  Stroke="White"
+                                  StrokeLineJoin="Round"
+                                  StrokeStartLineCap="Round"
+                                  StrokeEndLineCap="Round"
+                                  Data="M.256.255a.874.874 0 0 0-.18.974l4.753 17.114a.875.875 0 0 0 1.603-.012L10 10l8.334-3.57a.875.875 0 0 0 .01-1.601L1.23.075a.874.874 0 0 0-.974.18Z" />
+                        </Viewbox>
+                        <Border CornerRadius="3"
+                                Background="{Binding Color, Converter={StaticResource ColorToSolidColorBrushConverter}, ConverterParameter=0.7}"
+                                Padding="6 2">
+                            <TextBlock Text="{Binding Name}" />
+                        </Border>
+                    </StackPanel>
+                </DataTemplate>
+
+                <Style TargetType="{x:Type nodify:PendingConnection}"
+                       BasedOn="{StaticResource {x:Type nodify:PendingConnection}}">
+                    <Setter Property="Direction"
+                            Value="{Binding Source.Direction}" />
```

Comment:
Unless I am mistaken, DataContext here is `ConnectionViewModel`, which has `ConnectorViewModel Source { get; }` property, but the connector doesn't have `Direction` property.

Reply:
Removed (copy paste leftovers).